### PR TITLE
feat(cli): interactively set number of seeded items per seeder

### PR DESCRIPTION
**Pull request by jeabakker**
The seeders can now have their own default limit

### CODE REVIEW

File: engine/classes/Elgg/Cli/DatabaseSeedCommand.php [hunk 1]
```diff
@@ -63,12 +63,14 @@ protected function command() {
...
 
 		$options = [
-			'limit' => (int) $this->option('limit') ?: 20,
+			'limit' => $this->option('limit'),
 			'image_folder' => $this->option('image_folder'),
 			'type' => $this->option('type'),
 			'create_since' => $this->option('create_since'),
 			'create_until' => $this->option('create_until'),
 			'create' => (bool) $this->argument('create'),
+			'interactive' => !(bool) $this->option('no-interaction'),
```
Comment (jdalsem) [inline engine/classes/Elgg/Cli/DatabaseSeedCommand.php:72]: you can determine interactive also in the seeder if you have cli_command available

File: engine/classes/Elgg/Database/Seeds/Seed.php [hunk 2]
```diff
@@ -21,17 +21,17 @@ abstract class Seed implements Seedable {
 	/**
 	 * @var int Max number of items to be created by the seed
 	 */
-	protected $limit = 20;
+	protected int $limit = 20;
```
Comment (jdalsem) [inline engine/classes/Elgg/Database/Seeds/Seed.php:24]: you can also set this value to the defaultLimit function in the constructor

**Review by jdalsem — CHANGES_REQUESTED** [review]

File: engine/classes/Elgg/Cli/DatabaseSeedCommand.php (hunk 1 above)
Comment (jeabakker) [reply engine/classes/Elgg/Cli/DatabaseSeedCommand.php:72]: I'm already having trouble with passing the CLI command to the seeder service for interaction, so i don't want to move any more of the logic to that service.

Keeping this as is

File: engine/classes/Elgg/Database/Seeds/Seed.php (hunk 2 above)
Comment (jeabakker) [reply engine/classes/Elgg/Database/Seeds/Seed.php]: moved to constructor

**Review by jdalsem — APPROVED** [review]

**Pull request merged**